CMakeLists.txt

```diff
@@ -1,5 +1,5 @@
 # CMake project definition file.
-cmake_minimum_required(VERSION 3.3...3.5)
+cmake_minimum_required(VERSION 3.5)
```

Comment:
First of all, thank you for your review.

I updated the minimum because on CMake 3.28.1 it outputs a deprecation warning:

```
Compatibility with CMake < 3.5 will be removed from a future version of
CMake.

Update the VERSION argument <min> value or use a ...<max> suffix to tell
CMake that the project does not need compatibility with older versions.
```

Reply:
This warning appears when you use just "3.3", but when it's "3.2...3.5", it should work fine. Actually many other projects did such fix. Basically the range of minimal versions instead of explicit version means that you do maintain of such compatibility manually by providing conditional scopes that works differently depending on the CMake version. I got the same warning when it was just 3.3, but I fixed it once I set "3.3...3.5" range. The warning means that some of sub-projects had just "3.3". So, they need to be changed into the range.

Reply:
On my Mac is CMake 2.29.1, and there is no of such warning with thet current mainstream:
```
vitaly@whl-arm64-001 build % cmake ..
-- The C compiler identification is AppleClang 15.0.0.15000309
-- The CXX compiler identification is AppleClang 15.0.0.15000309
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Check for working C compiler: /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/cc - skipped
-- Detecting C compile features
-- Detecting C compile features - done
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Check for working CXX compiler: /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/c++ - skipped
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- VGM/GYM: Nuked OPN2 emulator will be used
-- Performing Test LIBM_NOT_NEEDED
-- Performing Test LIBM_NOT_NEEDED - Success
-- ZLib library located, compressed file formats will be supported
-- SDL2 library located, player demo is available to be built in the /player directory
-- ** UnRAR library not found, player demo will not support the RAR file format
-- Configuring done (1.2s)
-- Generating done (0.0s)
-- Build files have been written to: /Users/vitaly/Repos/game-music-emu/build
vitaly@whl-arm64-001 build % cmake --version
cmake version 3.29.1

CMake suite maintained and supported by Kitware (kitware.com/cmake).
```

Reply:
Yeah, from that message it seems like adding the range should solve it, but I'm still getting the warning. Maybe this is a bug in CMake or outdated message?

Reply:
Ye, looks like more like a bug. Try a bit newer version to see what happen.

Reply:
That's weird, I'm not sure what may have caused the initial message that led me to add 3.5. The problem here was that I added two periods instead of three. I'll revert the version to `3.3...3.5` then, sorry for the confusion.